rust/azure_iot_operations_services/src/azure_device_registry.rs

```diff
@@ -5,20 +5,179 @@
 
 //! Types for Azure Device Registry operations.
 
+use core::fmt::Debug;
+use std::collections::HashMap;
+
+use crate::azure_device_registry::device_name_gen::adr_base_service::client as adr_name_gen;
+
 /// Azure Device Registry generated code
 mod device_name_gen;
 
 // ~~~~~~~~~~~~~~~~~~~SDK Created Structs~~~~~~~~~~~~~~~~~~~~~~~~
 
 // ~~~~~~~~~~~~~~~~~~~Helper fns ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 fn option_vec_from<T, U>(source: Option<Vec<T>>, into_fn: impl Fn(T) -> U) -> Option<Vec<U>> {
-  source.map(|vec| vec.into_iter().map(into_fn).collect())
+    source.map(|vec| vec.into_iter().map(into_fn).collect())
 }
 
 // ~~~~~~~~~~~~~~~~~~~Common DTDL Equivalent Structs~~~~~~~~~~~~~
+#[derive(Clone, Debug)]
+/// Represents an error in the configuration of an asset or device.
+pub struct ConfigError {
+    /// The code of the error.
+    pub code: Option<String>,
+    /// Array of event statuses that describe the status of each event.
+    pub details: Option<Vec<Details>>,
+    /// The inner error, if any.
+    pub inner_error: Option<HashMap<String, String>>,
+    /// The message of the error.
+    pub message: Option<String>,
+}
+
+impl From<ConfigError> for adr_name_gen::ConfigError {
+    fn from(value: ConfigError) -> Self {
+        adr_name_gen::ConfigError {
+            code: value.code,
+            message: value.message,
+            details: option_vec_from(value.details, |details| {
+                adr_name_gen::DetailsSchemaElementSchema {
+                    code: details.code,
+                    correlation_id: details.correlation_id,
+                    info: details.info,
+                    message: details.message,
+                }
+            }),
+            inner_error: value.inner_error,
+        }
+    }
+}
+
+impl From<adr_name_gen::ConfigError> for ConfigError {
+    fn from(value: adr_name_gen::ConfigError) -> Self {
+        ConfigError {
+            code: value.code,
+            message: value.message,
+            details: option_vec_from(value.details, |details| Details {
+                code: details.code,
+                correlation_id: details.correlation_id,
+                info: details.info,
+                message: details.message,
+            }),
+            inner_error: value.inner_error,
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+/// Details describing an error?
```

Comment:
```suggestion
/// Details describing an error
```

Not sure if this is a typo

Reply:
we're not quite sure what this struct is for yet. I'll remove the doc comment as a whole for now